### PR TITLE
Remove next gen build cache prototype

### PR DESCRIPTION
**Pull request by asodja**
I left only `H2BuildCacheService` and it's interface `StatefulNextGenBuildCacheService`.

Fixes https://github.com/gradle/gradle/issues/26321

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test this

File: subprojects/execution/src/main/java/org/gradle/internal/execution/steps/ResolveCachingStateStep.java [hunk 1]
```diff
@@ -92,9 +91,8 @@ private CachingState calculateCachingState(UnitOfWork work, BeforeExecutionState
             .orElse(null);
         work.shouldDisableCaching(detectedOverlappingOutputs)
             .ifPresent(cachingDisabledReasonsBuilder::add);
-        String cacheSalt = NextGenBuildCacheController.isNextGenCachingEnabled() ? "next-gen" : null;
 
-        return cachingStateFactory.createCachingState(beforeExecutionState, cacheSalt, cachingDisabledReasonsBuilder.build());
+        return cachingStateFactory.createCachingState(beforeExecutionState, null, cachingDisabledReasonsBuilder.build());
```
Comment (lptr) [inline subprojects/execution/src/main/java/org/gradle/internal/execution/steps/ResolveCachingStateStep.java:95]: ❓ I guess we can remove this parameter entirely, no?

**Review by lptr — APPROVED** [review]
LGTM.

**Comment by asodja** [comment]
@bot-gradle test and merge

**Pull request merged**